x/merkledb/intermediate_node_db_test.go

```diff
@@ -97,24 +90,22 @@ func TestIntermediateNodeDB(t *testing.T) {
 	// Put one more element in the cache, which should trigger an eviction
 	// of all but 2 elements. 2 elements remain rather than 1 element because of
 	// the added key prefix increasing the size tracked by the batch.
-	key = newPath([]byte{byte(cacheSize)})
-	node := &node{
-		dbNode: dbNode{
-			value: maybe.Some([]byte{byte(cacheSize)}),
-		},
-	}
+	key = newPath([]byte{byte(added)})
+	node := newNode(nil, EmptyPath)
+	node.setValue(maybe.Some([]byte{byte(added)}))
 	require.NoError(db.Put(key, node))
 
 	// Assert cache has expected number of elements
-	require.Equal(2, db.nodeCache.fifo.Len())
+	require.Equal(1, db.nodeCache.fifo.Len())
 	gotKey, _, ok := db.nodeCache.fifo.Oldest()
 	require.True(ok)
-	require.Equal(newPath([]byte{byte(added - 1)}), gotKey)
+	require.Equal(newPath([]byte{byte(added)}), gotKey)
 
-	// Get a node from the base database (not cache)
-	nodeRead, err := db.Get(newPath([]byte{0x03}))
+	// Get a node from the base database
+	// Use an early key that has been evicted from the cache
+	nodeRead, err := db.Get(newPath([]byte{0x01}))
```

Comment:
Can we also assert that this isn't in the cache? Also should've done that in #1947 but missed that